src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -414,5 +430,59 @@ private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle,
                 }
             }
         }
+
+        private static void PollFunction(SafeCurlMultiHandle multiHandle)
+        {
+            List<Interop.libc.pollfd> fds = new List<Interop.libc.pollfd>();
+
+            while (true)
+            {
+                lock (multiHandle)
+                {
+                    if (0 == multiHandle.FdCount)
+                    {
+                        multiHandle.PollCancelled = true;
+                        return;
+                    }
+                }
+
+                multiHandle.PollFds(fds);
+                if (0 == fds.Count)
+                {
+                    // ignore errors
```

Comment:
Could we end up in some kind of infinite loop here where multiHandle.FdCount > 0 but we get an error every time we call PollFds, so we end up in a tight spin loop?